contrib/google.golang.org/grpc.v12/grpc.go

```diff
@@ -43,23 +43,27 @@ func UnaryServerInterceptor(opts ...InterceptorOption) grpc.UnaryServerIntercept
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		span, ctx := startSpanFromContext(ctx, info.FullMethod, cfg.serviceName, cfg.analyticsRate)
 		resp, err := handler(ctx, req)
+		span.SetTag(ext.GRPCStatus, grpc.Code(err))
 		span.Finish(tracer.WithError(err))
 		return resp, err
 	}
 }
 
 func startSpanFromContext(ctx context.Context, method, service string, rate float64) (ddtrace.Span, context.Context) {
+	rpcTags := grpcutil.ExtractRPCTags(method)
 	opts := []ddtrace.StartSpanOption{
 		tracer.ServiceName(service),
 		tracer.ResourceName(method),
-		tracer.Tag(tagMethod, method),
 		tracer.SpanType(ext.AppTypeRPC),
 		tracer.Measured(),
 		tracer.Tag(ext.Component, "google.golang.org/grpc.v12"),
 		tracer.Tag(ext.SpanKind, ext.SpanKindServer),
-		tracer.Tag(ext.RPCMethod, method),
-		tracer.Tag(ext.RPCSystem, "gRPC"),
-		tracer.Tag(ext.RPCService, service),
+		tracer.Tag(ext.RPCSystem, "grpc"),
+		tracer.Tag(ext.RPCService, rpcTags.Service),
+		tracer.Tag(ext.RPCMethod, rpcTags.Method),
+		tracer.Tag(ext.GRPCPackage, rpcTags.Package),
```

Comment:
Do we want to set these even if they're empty or "unknown" strings?

Reply:
> Realistically, the only way this ever returns empty is if theres an error, so it shouldn't be a problem

So `ExtractRPCTags` should return the error? bc right now it returns empty or "unknown" strings in case of error?

Reply:
Not really, because `ExtractRPCTags` simply extracts whatever values are provided to it. It should never reach this point without an error in the RPC configuration itself. If somehow, someway the RPC method is empty or improperly formatted, `ExtractRPCTags` will return whatever is given to it, but I don't think its functionality is to determine whether the GRPC is properly configured. That would fall somewhere else in the integration if we do decide to support it. Essentially this part of the code is purely informational based on what is passed into it

Reply:
```golang
	if len(elems) < 2 { //Empty string check
		tags.Method = ""
		tags.Service = ""
		tags.Package = ""
		return tags
	} else if len(elems) > 2 { //Improper grpc fullmethod check
		tags.Method = "unknown"
		tags.Service = "unknown"
		tags.Package = "unknown"
		return tags
	}
```
> Not really, because ExtractRPCTags simply extracts whatever values are provided to it

Doesn't look like it does, based on the code snippet.

> It should never reach this point without an error in the RPC configuration itself.

Why would we check the length in that case?
I'm not saying we shouldn't check the length, I'm actually in favor of always anticipating unexpected scenarios. 

> ExtractRPCTags will return whatever is given to it, but I don't think its functionality is to determine whether the GRPC is properly configured.

I'm just asking to check for that error and avoid putting empty and/or "unknown" strings as tags if the error is not nil.

Reply:
I'm in agreement with @ahmed-mez that this could be a bit more error-resistant. I made a suggestion for refactoring that method inline. At that point, we can just only set those tags if the error is not nil. e.g. this code would look like this:

```

	opts := []ddtrace.StartSpanOption{
		tracer.ServiceName(service),
		tracer.ResourceName(method),
		....
		tracer.Tag(ext.GRPCFullMethod, method),
		tracer.Tag(ext.GRPCKind, "unary"),
		}
	if rpcTags, err := grpcutil.ExtractRPCTags(method); err != nil {
		opts = append(opts, tracer.Tag(ext.RPCService, rpcTags.Service),
			tracer.Tag(ext.RPCMethod, rpcTags.Method),
			tracer.Tag(ext.GRPCPackage, rpcTags.Package))
	}
```
Or something similar...

Does that make sense @zhamza99? Also happy to just discuss this offline, in-person tomorrow if that'd be helpful.